docs/ParameterValidation.md

```diff
@@ -0,0 +1,177 @@
+# Helium Parameter Validation Documentation
+
+The goal of this document is to define valid/invalid APIs and Query parameters and provide specifications on resulting behavior of Helium API.
+
+## Common Status code rules followed for all APIs
+
+### Searches
+
+- Valid input returns 200 with JSON Array [{},{}...]
+- Valid input with no results returns 200 with empty array
+- Invalid input returns 400 text/plain with Invalid error response
+
+### Single Reads
+
+- Valid single reads returns 200 with JSON document
+- Valid single reads with no results returns 404 with No {entity} found error response
+- Invalid single reads like /api/movies/tt1234,/api/actors/nm1234 returns 400 with Invalid error response
+
+### Multiple Error Scenarios
+
+- Fail on first error
+
+## Movies
+
+### API /api/movies
+
+### Movies Query Parameters
+
+- Name: q (search)
+- Description: This is a case insensitive search query (q) string on movie title or actor name field
+- Type: string
+- Parameter Validation:
+  - Valid input: query length >= 3 non-whitespace characters
+    - Example: 'ring','Ring'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by search query on title or actor name field as [{},{},..],empty array when no results
+  - Invalid input: query length < 3 characters
```

Comment:
Small nit change - in line 37 right before "empty array" and a few other places below, there are commas without spaces after.  Not a big deal, but was a little weird to read.  Maybe do one more scrub for spaces after commas and other spacing.

Reply:
updated!